res/controllers/common-controller-scripts.js

```diff
@@ -152,9 +152,9 @@ script.midiDebug = function(channel, control, value, status, group) {
 // Returns the deck number of a "ChannelN" or "SamplerN" group
 script.deckFromGroup = function(group) {
     var deck = 0;
-    if (group.substring(2, 8) === "hannel") {
-        // Extract deck number from the group text
-        deck = group.substring(8, group.length - 1);
+    if (group.match("Channel.")) {
```

Comment:
Evaluating a regular expressions is overkill where extracting a substring would be sufficient. Moreover it is inconsistent with the following substring extraction based on this evaluation.

Reply:
But to fix the bug just extracting a substring is not sufficient.
The inconsistency is a good point and easy to remove.

Reply:
I don't know how much of an issue that is for just a few midi events.
However, one does not need to use this helper function in case of performance issues and use substring directly.

Reply:
This function is used by official mappings and should not be slower for them.
What about using `if else {` to check for other group patterns like `[EffectRack1_...[ChannelN]..`?

Reply:
Possible, but I don't have an overview of all existing groups and groups are changing.
Will it be so much faster?

Reply:
I could make a notable difference I think, especially for controller mappings that are latency critical.

I'm also not really sure that this change is needed. It seems like a special case and I'd rather have a special function for that (`deckFromEffectGroup`). Also, since we're redesigning the JS APIs of the new controller system from the ground up, this will only make a difference for legacy controller mappings.

Reply:
I see. When will this new system become available?

Reply:
Mixxx 2.4 or 2.5 probably.